unicode/src/main/java/com/google/re2j/UnicodeTablesGenerator.java

```diff
@@ -0,0 +1,446 @@
+package com.google.re2j;
+
+import com.google.common.collect.Sets;
+import com.google.common.collect.SortedSetMultimap;
+import com.google.common.collect.TreeMultimap;
+import com.google.googlejavaformat.java.Formatter;
+import com.google.googlejavaformat.java.FormatterException;
+import com.ibm.icu.impl.UPropertyAliases;
```

Comment:
Is the icu dependency necessary? What does Character.getType not provide?

Reply:
I don't think there is any case folding information provided in java.lang.Character, but maybe I'm wrong.

Reply:
Character.toUpperCaseEx will case-fold a single code point. (Of course, it doesn't properly normalize because that's a function from strings to strings.)

Reply:
Thank you, one learns something every day!

I also couldn't find a source for unicode category names (e.g. 15 -> "Cc") in Java.

Reply:
Generating the tables at startup would be excellent, it would solve the problem of mismatched RE2/J <-> JDK data.

I'll modify the impl to remove the ICU4J dependency. Thanks for the pointer.